src/main/java/com/parasoft/findings/sonar/sensor/CoverageSensor.java

```diff
@@ -26,9 +43,57 @@ public void describe(SensorDescriptor sensorDescriptor) {
     public void execute(SensorContext sensorContext) {
         String[] reportPaths = sensorContext.config().getStringArray(ParasoftConstants.PARASOFT_COVERAGE_REPORT_PATHS_KEY);
         if (reportPaths != null) {
-            for (String reportPath : reportPaths) {
-                Logger.getLogger().info("ReportPath = " + reportPath);
+            List<File> coberturaReports = transformToCoberturaReports(reportPaths);
+            for (File coberturaReport : coberturaReports) {
+                Logger.getLogger().info(NLS.bind(Messages.UploadCodeCoverageData, coberturaReport.getName()));
+            }
+        }
+    }
+
+    public List<File> transformToCoberturaReports (String[] reportPaths) {
+        List<File> coberturaReports = new ArrayList<>();
+
+        for (String reportPath : reportPaths) {
+            File reportFile = new File(fs.baseDir(), reportPath);
+            if (!reportFile.isFile() || !reportFile.exists() || !reportFile.canRead()) {
+                Logger.getLogger().warn(NLS.bind(Messages.InvalidReportFile, reportFile.getAbsolutePath()));
+                continue;
+            } else {
+                Logger.getLogger().info(NLS.bind(Messages.ParsingReportFile, reportFile.getName()));
+                File resultFile = transformToCoberturaFormat(reportFile);
+                if (resultFile != null) {
+                    coberturaReports.add(resultFile);
+                }
             }
         }
+        return coberturaReports;
+    }
+
+    public File transformToCoberturaFormat(File report) {
+        try {
+            File result = new File(report.getAbsolutePath() + "-cobertura.xml");
+
+            Source xsltFile = new StreamSource(getClass().getResourceAsStream("/com/parasoft/findings/sonar/res/xsl/cobertura.xsl"));
+            Processor processor = new Processor(false);
+            XsltCompiler compiler = processor.newXsltCompiler();
+            XsltExecutable stylesheet = compiler.compile(xsltFile);
+            Xslt30Transformer transformer = stylesheet.load30();
+
+            Serializer out = processor.newSerializer(result);
+            out.setOutputProperty(Serializer.Property.METHOD, "xml");
+            out.setOutputProperty(Serializer.Property.INDENT, "yes");
+
+            Map<QName, XdmValue> paramsMap = new LinkedHashMap<>();
+            QName paramName = new QName("pipelineBuildWorkingDirectory");
+            XdmValue paramValue = new XdmAtomicValue(fs.baseDir().getAbsolutePath());
+            paramsMap.put(paramName, paramValue);
+            transformer.setStylesheetParameters(paramsMap);
+
+            transformer.transform(new StreamSource(report), out);
+            return result;
+        } catch (Exception e) {
+            Logger.getLogger().error(e.getMessage(), e);
```

Comment:
Give better error message instead of default message from the exception.
For example:
Change "e.getMessage()" to something like "Failed to transform file: xxx";